### DISCUSSION TITLE
Pep8 formatting for malt?

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#345] [Development]
c34e0b9a0c624358268c8c9d922ca16fe1c2a04e I formatted BlenderMalt to follow pep8 standards as a test. You can use VSCode to autoformat your code, there are different options for it, like manual formatting and formatting on save ect. How do you like it? Would that be a good standard to adapt for Malt to save some time manually reformatting code from contributers?

**Comment by pragma37** [comment]
I think it's a good idea to enforce some kind of auto-formatting, but I'm not sure this is the right one.
Is this autopep8?

It seems a bit too opinionated and somewhat inconsistent in some places:
https://github.com/bnpr/Malt/commit/c34e0b9a0c624358268c8c9d922ca16fe1c2a04e#diff-664c5a744c0342f732e0fff546dfb2dd5dadf94d6a740ed5731d277d944e4b54L64

I also would bet nobody is doing graphics programming in some kind of legacy terminal with a 79 characters limit. 😅

**Reply by Kolupsy** [reply]
yes this is autopep8. VSCode provides a few default formatters:
![image](https://user-images.githubusercontent.com/88381426/177608604-72bbabd3-d162-4eb6-95a8-4cba82f28b17.png)

> It seems a bit too opinionated and somewhat inconsistent in some places

I does have to follow some conventions and I noticed too how it can do slightly unexpected formatting but the question is if you would rather manually reformat the code of contributers than using some sort of standard, even if its not perfect. 
I think you could also configure your own ruleset for formatting.

**Comment by pragma37** [comment]
I've been testing the available options and I find all of them do more harm than good.

I'd like something that handles things like max line length, trailing whitespaces or blank lines, but still leaves room for the programmer to format their code.

yapf has many options but all of them must be enforced in some way or another, you can't just tell the formatter to do nothing in certain cases.

So, for now, I'd say just follow the *when in Rome, do as the romans* rule.
If you find a format rule followed across the codebase then stick to it, otherwise do whatever you find more readable.

**Reply by Kolupsy** [reply]
alright then lets leave it as we were. I will try to kind of stick with the style that you have already been doing thoughout the whole code base, obviously readability it somewhat subjective but I get OCD if some of the code looks a lot different from the rest so this is what I will do 😅
